### PR TITLE
Deploy s3 bundle step and update app-manifest - GH action CI workflow

### PR DESCRIPTION
**Pull request by ojasjoshi**
*Issue #, if available:*

*Description of changes:*
Adding s3 bundle deployment and manifest update to scheduled ros1 ros2 GH workflows

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ojasjoshi** [comment]
Closing this PR till update-manifest GH action is merged.

**Review by wangvnn — APPROVED** [review]

**Comment by ojasjoshi** [comment]
Note: This PR is to be left unmerged before some related issues are discussed/resolved.

File: .github/workflows/ros1.yml [hunk 1]
```diff
@@ -56,6 +56,37 @@ jobs:
...
+     - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_ROS1 }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_ROS1 }}
+        aws-region: ${{ secrets.AWS_REGION }}
+      if: github.event_name == 'schedule' && github.ref == 'refs/heads/ros1'
+    - id: deploy_bundle
+      name: Deploy bundle to S3
+      uses: aws-robomaker-github-actions/s3-cp-action@v2.1.1
```
Comment (mm318) [inline .github/workflows/ros1.yml]: Shouldn't this be `aws-robotics/aws-robomaker-github-actions/s3-cp-action@2.1.1`?

File: .github/workflows/ros1.yml [hunk 2]
```diff
@@ -56,6 +56,37 @@ jobs:
...
+        id: time
+        run: echo "::set-output name=timestamp::$(date +%s)"
+    - name: Update App-manifest version number
+      id: update-app-manifest
+      uses: aws-robotics/robomaker-sample-app-ci/codecommit-put-file-action@v2.1.1
+      env:
+        REPO_NAME: 'AppManifest-hello-world-${{matrix.distro}}-gazebo${{matrix.gazebo}}'
+        BRANCH_NAME: 'mainline'
+        FILE_CONTENT: '{\"application_version\": \"${{ steps.robot_ws_build.outputs.sample-app-version }}.${{ github.run_number }}\", \"timestamp\":\"${{ steps.time.outputs.timestamp }}\"}'
+        FILE_PATH: '/version.json'
```
Comment (mm318) [inline .github/workflows/ros1.yml]: These will need to be `DEST_FILE_CONTENT` and `DEST_FILE_PATH` now, right?

**Review by mm318 — COMMENTED** [review]
The comments I made to `ros1.yml` applies to `ros2.yml` as well.

**Comment by mm318** [comment]
In general, I think this change needs more testing.

In order to test the trigger by `schedule` case, I would make a fork of this repository and set its GitHub secrets to point to your own non-production AWS account that has the necessary S3 buckets and CodeCommit repos setup.

**Comment by mm318** [comment]
Also, looking at all the checks that have passed, I'm concerned that `Build and Bundle (ROS1)` has not shown up, and that the Travis ones are still there, which means we'll get conflicting bundle uploads to S3 (one from Travis and one from GitHub actions, assuming it is working).

File: .github/workflows/ros1.yml (hunk 2 above)
Comment (ojasjoshi) [reply .github/workflows/ros1.yml]: Yep good catch.. made the changes accordingly

File: .github/workflows/ros1.yml [hunk 3]
```diff
@@ -43,19 +43,50 @@ jobs:
...
         ros-distro: ${{ matrix.distro }}
         gazebo-version: ${{ matrix.gazebo }}
         workspace-dir: simulation_ws
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_ROS1 }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_ROS1 }}
+        aws-region: ${{ secrets.AWS_REGION }}
+      if: github.event_name == 'schedule' &&  ${{ ! contains(steps.robot_ws_build.outcome, 'failure') }} & ${{ ! contains(steps.simulation_ws_build.outcome, 'failure') }}
```
Comment (mm318) [inline .github/workflows/ros1.yml]: I'm not sure what went wrong earlier, but my guess is this should be like:
```suggestion
      if: ${{ github.event_name == 'schedule' && !contains(steps.robot_ws_build.outcome, 'failure') && !contains(steps.simulation_ws_build.outcome, 'failure') }}
```

**Review by mm318 — COMMENTED** [review]
Also, reminder to bump the minor version in `package.xml` and remove the `.travis.yml` (so that we don't have Travis CI and GitHub Actions uploading bundles at the same time).

File: .github/workflows/ros1.yml (hunk 3 above)
Comment (ojasjoshi) [reply .github/workflows/ros1.yml]: yep trying if that helps.. wording in the [documentation](https://docs.github.com/en/articles/workflow-syntax-for-github-actions/#jobsjob_idif) is a bit ambiguous.

**Review by mm318 — APPROVED** [review]

**Pull request merged**